src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -263,15 +257,35 @@ def evaluate_workflows_action_filters(
         extra={
             "group_id": event_data.group.id,
             "event_id": event_id,
-            "workflow_ids": [workflow.id for workflow in workflows],
-            "action_conditions": [action_condition.id for action_condition in action_conditions],
+            "workflow_ids": list(workflows_by_id.keys()),
+            "action_conditions": [
+                action_condition.id for action_condition in dcg_to_workflow_id.keys()
+            ],
             "filtered_action_groups": [action_group.id for action_group in filtered_action_groups],
         },
     )
 
     return filtered_action_groups
 
 
+@sentry_sdk.trace
+def evaluate_workflows_action_filters(
```

Comment:
this is a bit confusing, how will we know when to call `evaluate_workflows_action_filter` compared to `evaluate_action_filters`? (the names are so similar it's a bit tricky to grok).

mind adding some comments / information around this? if we're not expecting to call `evaluate_action_filters` anymore, should we keep that method?

Reply:
I'm trying to share logic where possible. The N+1 comes from the repeated queries to `DataConditionGroup` in `evaluate_workflows_action_filters` because we have different events to process in delayed workflow, but we can avoid that by doing a single call to `DataConditionGroup` over all the events to process and then passing in the relevant data into the factored out function (`evaluate_action_filters`). Let me think how to make it more obvious that that's what it's doing